File: Linguagens de Programação/Tarefa1 Java/Pilha.java

public class Pilha {
	
	private int[] stack;
	private int max;
	private int top;
	
	public Pilha(int tam){
		
		this.max = tam - 1;
		this.top = -1; // pilha vazia
		this.stack = new int[tam];
	}
	
	public void Push(int x){
		
		if(this.top < this.max){
			this.top += 1;
			this.stack[this.top] = x;
			
		}
		else{
			System.out.println("Pilha Cheia");
		}
			
		
	}
	
	public void top(){
		if(this.top > -1){
			System.out.println(this.stack[this.top]);
		}
		else{
			System.out.println("Pilha vazia");
		}
		
	}
}

File: Algoritmo/code_study/example_08/pilha.c
#include <stdio.h>
#include "pilha.h"
#define TAM 10

/*inicializa um pilha vazia*/
void inicializa(Pilha *p){

	int i;
	int a = -1;	
	p->topo = a;
	for(i=0;i<TAM;i++){
		p->pilha[i] = p->topo;
	}
	

}

/*visualiza uma pilha vazia*/
void ver(Pilha *p){

		
	int i;
	p->pilha[TAM] = p->pilha[TAM];
	for(i=0;i<TAM;i++){
		printf("%d\n",p->pilha[i]);
	}

	printf("-------------------------------\n");
		
}

/*insere um inteiro no topo da pilha*/
void insere(Pilha *p, int elem){

	p->topo = elem;
	p->pilha[0] = elem;

}

/*retira um elemento da pilha*/
void retira(Pilha *p){

	p->pilha[0] = -1;
}

/*imprime o elemento do topo da pilha*/
void imprimetopo(Pilha *p){

	printf("Topo : %d\n",p->topo);

}


